src/caches/beliefpropagationcache.jl

```diff
@@ -16,11 +16,11 @@ using SimpleTraits: SimpleTraits, Not, @traitfn
 default_message(inds_e) = ITensor[denseblocks(delta(i)) for i in inds_e]
 default_messages(ptn::PartitionedGraph) = Dictionary()
 default_message_norm(m::ITensor) = norm(m)
-function default_message_update(contract_list::Vector{ITensor}; kwargs...)
+function default_message_update(contract_list::Vector{ITensor}; normalize=true, kwargs...)
   sequence = optimal_contraction_sequence(contract_list)
   updated_messages = contract(contract_list; sequence, kwargs...)
   message_norm = norm(updated_messages)
-  if !iszero(message_norm)
+  if !iszero(message_norm) && normalize
```

Comment:
```suggestion
  if normalize && !iszero(message_norm)
```
so then it only checks if it is zero if you try to normalize (slightly more efficient).